### PR TITLE
Use orElseGet() to avoid unnecessary object allocation

### PR DESCRIPTION
**Pull request by sabi0**

### CODE REVIEW

**Review by uschindler — CHANGES_REQUESTED** [review]
Most of the changes are fine.

But for some of the code it is really not an issue:
`VectorizationProvider` is executing this code only once. The actual method call returns a constant so there's no object allocation involved. As the code is only executed once, the overhead by the invokeDynamic and the dynamically allocated functional class is making that code actually slower and making Lucene's static initialization worse.

File: lucene/core/src/java/org/apache/lucene/internal/vectorization/VectorizationProvider.java [hunk 1]
```diff
@@ -185,7 +185,7 @@ static VectorizationProvider lookup(boolean testMode) {
    */
   private static Optional<Module> lookupVectorModule() {
     return Optional.ofNullable(VectorizationProvider.class.getModule().getLayer())
-        .orElse(ModuleLayer.boot())
```
[the remark this answers is not in the document]
Comment (sabi0) [reply lucene/core/src/java/org/apache/lucene/internal/vectorization/VectorizationProvider.java:188]: Thank you for the insightful comments.
I will remove the unwanted changes.

**Comment by sabi0** [comment]
I've added my name to the GitHub profile. And will review the PRs to describe the reasoning.

-Dmitry

**Review by uschindler — APPROVED** [review]

**Comment by uschindler** [comment]
It would be nice to have a changes entry in `CHANGES.txt`. You can add it to 9.10 section and I will cherry pick the changes in Lucene 9.x.

As you have more open PRs with cleanups, it would be good to have a generic one and we lost all your PR numbers there.

**Comment by uschindler** [comment]
Looks fine. Have you added all PR numbers that were merged today and the ones I approved already?  I will merge them soon.

**Pull request merged**